### DISCUSSION TITLE
Doubt on how to use webview

### DISCUSSION THREAD

**Discussion by Jagathishrex** [microsoft/vscode-discussions#970] [Extension Development QnA] [answered]
I am creating an webview extenstion, that will load my webpage content as iframe. 

The content will be only loaded if the user is signed ip to my website.

If the user is not signed-in we will internally redirect to other page through JSP. 

The redirected network call are displayed in the debug tab. But the I frame doesn't display the redirected webpage. 

<img width="1022" alt="Screenshot 2023-12-08 at 11 24 46 PM" src="https://github.com/microsoft/vscode-discussions/assets/22112325/556cbd12-6d6c-4133-ae7d-1780aa76cb05">

In the above image, you can see the iframe is blank and the request to load the resource returns 302.

**Comment by r3m0t** [comment, marked as answer]
Check the Console tab as this might violate a built in VS Code CSP policy. Is the localnila .com in an `<iframe>` created by your extension, or is it in an `<iframe>` that VS Code created?

**Reply by Jagathishrex** [reply]
localnila .com is created by my extension. 

That is returned from the below function. 

```
function getWebviewContent() {
  return `
            <!DOCTYPE html>
            <html lang="en">
            <head>
                <meta charset="UTF-8">
                <meta name="viewport" content="width=device-width, initial-scale=1.0">
                <title>Example Webview</title>
            </head>
            <body>
                <iframe src="https://stageclient.localnila.com/view/space/77166659/project/838c9d74-ddec-4894-8bc9-9079fe8a37a4/document/F69EB88A-EB33-4F49-9371-267C4586DE90" type="link"></iframe>
            </body>
            </html>
        `;
}
```
<img width="1025" alt="Screenshot 2023-12-08 at 11 40 14 PM" src="https://github.com/microsoft/vscode-discussions/assets/22112325/7224d7cf-c1ae-4ceb-909c-6a5aec9ed162">

I have seen figma handled this case by opening their webpage to sign-in the user. Then load their project in vscode. 

Do you know what is the steps to do that ?

**Reply by r3m0t** [reply]
Normally it's done with `registerUriHandler`, `asExternalUri` and `openExternal`. Take a look at the documentation for `asExternalUri`.

**Reply by Jagathishrex** [reply]
Hi @r3m0t, Do you know any authentication provider library to handle the above case ? 
Because, I feel confused on creating it. 

Thanks.

**Reply by r3m0t** [reply]
There's an example in `vscode-extension-samples` repository. I would suggest creating a toy server with toy tokens, then when you get that working you will know how to implement the real one.

**Reply by Jagathishrex** [reply]
Let's assume, I have done the authentication. 
Is the session will be set automatically to my webview ?
